uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -726,135 +757,112 @@ define([
         vm.loading['fromDayTypes'] = false;
+
         return $q.reject('Please change date as it is not in any absence period');
       }
 
-      return $q.resolve(true);
+      return $q.resolve(previousAbsencePeriodId !== vm.period.id);
     }
 
-    /**
-     * Converts given date to javascript date as expected by uib-datepicker
-     *
-     * @param {String} date from server
-     * @return {Date}
-     */
-    function _convertDateFormatFromServer (date) {
-      return moment(date, sharedSettings.serverDateFormat).toDate();
-    }
-
-    /**
-     * Converts given date to server format
-     *
-     * @param {Date} date
-     * @return {String} date converted to server format
-     */
-    function _convertDateToServerFormat (date) {
-      return moment(date).format(sharedSettings.serverDateFormat);
-    }
-
-    /**
-     * Initializes user's calendar (work patterns)
-     *
-     * @return {Promise}
-     */
-    function _loadCalendar () {
-      return Calendar.get(vm.request.contact_id, vm.period.start_date, vm.period.end_date)
-        .then(function (usersCalendar) {
-          vm.calendar = usersCalendar;
-        });
-    }
-
-    /**
-     * Sets dates (and times in case of "hours" absence type) for vm.request from UI
-     */
-    function _setDates () {
+    function setDatesToRequest () {
       var options = vm.uiOptions;
       var request = vm.request;
-      var times;
+      var times = options.times;
 
-      request.from_date = options.fromDate ? vm._convertDateToServerFormat(options.fromDate) : null;
-      request.to_date = options.toDate ? vm._convertDateToServerFormat(options.toDate) : null;
+      request.from_date = options.fromDate ? convertDateToServerFormat(options.fromDate) : null;
+      request.to_date = options.toDate ? convertDateToServerFormat(options.toDate) : null;
 
       if (isCalculationUnit('hours') && !isLeaveType('toil')) {
-        times = options.times;
         request.from_date = request.from_date && times.from.time ? request.from_date + ' ' + times.from.time : null;
         request.to_date = request.to_date && times.to.time ? request.to_date + ' ' + times.to.time : null;
-        request.from_date_amount = !isNaN(+times.from.amount) ? times.from.amount : null;
-        request.to_date_amount = !isNaN(+times.to.amount) ? times.to.amount : null;
       }
 
       if (!options.multipleDays && options.fromDate) {
-        options.toDate = options.fromDate;
         request.to_date = request.from_date;
+        request.to_date_type = request.from_date_type;
+        options.toDate = options.fromDate;
       }
     }
 
     /**
-     * Sets dates and types for vm.request from UI
+     * Sets dates (with time in case of "hours" Absence Type) from UI to vm.request
+     *
+     * @param  {String} dateType from|to
+     * @return {Promise}
      */
-    function _setDateAndTypes () {
-      vm._setDates();
+    function setDatesFromUI (dateType) {
```

Comment:
why do we have to specify that this has anything to do with the UI? This should be a function being called generically when a date has been changed, correct? So in theory it could be called even if for some reason the dates changed internally

Reply:
Done.